quinn-udp/src/windows.rs

```diff
@@ -0,0 +1,160 @@
+        socket.0.set_nonblocking(true)?;
+        let addr = socket.0.local_addr()?;
+        let is_ipv6 = addr.as_socket_ipv6().is_some();
+        let v6only = unsafe {
+            let mut result: u32 = 0;
+            let mut len = mem::size_of_val(&result) as i32;
+            let rc = WinSock::getsockopt(
+                socket.0.as_raw_socket() as _,
+                WinSock::IPPROTO_IPV6,
+                WinSock::IPV6_V6ONLY as _,
+                &mut result as *mut _ as _,
+                &mut len,
+            );
+            if rc == -1 {
+                return Err(io::Error::last_os_error());
+            }
+            result != 0
+        };
+        let is_ipv4 = addr.as_socket_ipv4().is_some() || !v6only;
+
+        let sock_true: u32 = 1;
+
+        if is_ipv4 {
+            let rc = unsafe {
+                WinSock::setsockopt(
+                    socket.0.as_raw_socket() as _,
+                    WinSock::IPPROTO_IP as _,
+                    WinSock::IP_DONTFRAGMENT as _,
+                    &sock_true as *const _ as _,
+                    mem::size_of_val(&sock_true) as _,
+                )
+            };
+            if rc == -1 {
+                return Err(io::Error::last_os_error());
+            }
+        }
+
+        if is_ipv6 {
+            let rc = unsafe {
+                WinSock::setsockopt(
+                    socket.0.as_raw_socket() as _,
+                    WinSock::IPPROTO_IPV6 as _,
+                    WinSock::IPV6_DONTFRAG as _,
+                    &sock_true as *const _ as _,
+                    mem::size_of_val(&sock_true) as _,
+                )
+            };
+            if rc == -1 {
+                return Err(io::Error::last_os_error());
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn send(
+        &mut self,
+        socket: UdpSockRef<'_>,
+        _state: &UdpState,
+        transmits: &[Transmit],
+    ) -> Result<usize, io::Error> {
+        let mut sent = 0;
+        for transmit in transmits {
+            match socket.0.send_to(
+                &transmit.contents,
+                &socket2::SockAddr::from(transmit.destination),
+            ) {
+                Ok(_) => {
+                    sent += 1;
+                }
+                // We need to report that some packets were sent in this case, so we rely on
+                // errors being either harmlessly transient (in the case of WouldBlock) or
+                // recurring on the next call.
+                Err(_) if sent != 0 => return Ok(sent),
+                Err(e) => {
+                    if e.kind() == io::ErrorKind::WouldBlock {
+                        return Err(e);
+                    }
+
+                    // Other errors are ignored, since they will ususally be handled
+                    // by higher level retransmits and timeouts.
+                    // - PermissionDenied errors have been observed due to iptable rules.
```

Comment:
`s/iptable/firewall/`?

Reply:
Deleted the Linux-specific details here entirely.